### PR TITLE
Add sv tasks

### PR DESCRIPTION
**Pull request by cagaser**

### CODE REVIEW

File: bcftools.wdl [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    }
+    
+    command <<<
+        set -e
+        mkdir -p $(dirname ~{outputPath})
+        bcftools view ~{bcf} -O v -o ~{outputPath}
+    >>>
+    
+    output {
+        File OutputVcf = "~{outputPath}"
```
Comment (JasperBoom) [inline bcftools.wdl]: You don't need to add "~{}" for variables declared in the input when using them in the output.

File: bcftools.wdl [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+task Bcf2Vcf {
+    input {
+        File bcf
+        String outputPath
+        String dockerImage = "quay.io/biocontainers/bcftools:1.9--ha228f0b_3"
+    }
+    
+    command <<<
+        set -e
+        mkdir -p $(dirname ~{outputPath})
```
Comment (JasperBoom) [inline bcftools.wdl]:
```suggestion
        mkdir -p "$(dirname ~{outputPath})"
```

File: clever.wdl [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        File bamFile
+        File bamIndex
+        BwaIndex bwaIndex
+        String outputPath        
+        Int threads = 10 
+        Int mem = 15
+        String dockerImage = "quay.io/biocontainers/clever-toolkit:2.4--py36hcfe0e84_6"
+    }   
+    
+
```
Comment (JasperBoom) [inline clever.wdl]:
```suggestion
```

File: clever.wdl [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import "common.wdl"
+import "bwa.wdl"
+task Prediction {
+    input {
+        File bamFile
+        File bamIndex
+        BwaIndex bwaIndex
+        String outputPath        
+        Int threads = 10 
+        Int mem = 15
```
Comment (JasperBoom) [inline clever.wdl]:
```suggestion
        String memory = "15G"
```

File: clever.wdl [hunk 5]
```diff
@@ -0,0 +1,82 @@
+version 1.0 
+
+import "common.wdl"
+import "bwa.wdl"
```
Comment (JasperBoom) [inline clever.wdl]: What are these imports for?

File: clever.wdl [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        String outputPath        
+        Int threads = 10 
+        Int mem = 15
+        String dockerImage = "quay.io/biocontainers/clever-toolkit:2.4--py36hcfe0e84_6"
+    }   
+    
+
+    command <<< 
+        set -e
+        mkdir -p $(dirname ~{outputPath})
```
Comment (JasperBoom) [inline clever.wdl]:
```suggestion
        mkdir -p "$(dirname ~{outputPath})"
```

File: clever.wdl [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        --use_mapq \
+        --sorted \
+        -f \
+        ~{bamFile} \
+        ~{bwaIndex.fastaFile} \
+        ~{outputPath}
+    >>> 
+
+    output {
+        File predictions = "~{outputPath}/predictions.vcf"
```
Comment (JasperBoom) [inline clever.wdl]: See comment above.

File: clever.wdl [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    }   
+    
+    runtime {
+        cpu: threads
+        memory: mem
```
Comment (JasperBoom) [inline clever.wdl]:
```suggestion
        memory: memory
```

File: clever.wdl [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+task Mateclever {
+    input {
+        File fiteredBam
+        File indexedFiteredBam
+        BwaIndex bwaIndex
+        File predictions
+        String outputPath
+        Int threads = 10 
+        Int mem = 15
```
Comment (JasperBoom) [inline clever.wdl]:
```suggestion
        String memory = "15G"
```

File: clever.wdl [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+        Int cleverMaxDelLength = 100000
+        Int maxLengthDiff= 30
+        Int maxOffset = 150 
+        String dockerImage = "quay.io/biocontainers/clever-toolkit:2.4--py36hcfe0e84_6"
+    }
+
+    command <<<
+        set -e
+        mkdir -p $(dirname ~{outputPath})
```
Comment (JasperBoom) [inline clever.wdl]:
```suggestion
        mkdir -p "$(dirname ~{outputPath})"
```

File: clever.wdl [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+        -M ~{cleverMaxDelLength} \
+        -z ~{maxLengthDiff} \
+        -o ~{maxOffset} \
+        ~{bwaIndex.fastaFile} \
+        predictions.list \
+        ~{outputPath}
+    >>>
+    
+    output {
+        File matecleverVcf = "~{outputPath}/deletions.vcf"
```
Comment (JasperBoom) [inline clever.wdl]: See comment above.

File: clever.wdl [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+        ~{outputPath}
+    >>>
+    
+    output {
+        File matecleverVcf = "~{outputPath}/deletions.vcf" 
+    }
+    
+    runtime {
+        cpu: threads
+        memory: mem
```
Comment (JasperBoom) [inline clever.wdl]:
```suggestion
        memory: memory
```

File: delly.wdl [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+import "common.wdl"
+
+task CallSV {
+    input {
+        File bamFile
+        File bamIndex
+        File referenceFasta
+        File referenceFastaFai
+        String outputPath        
+        Int mem = 15
```
Comment (JasperBoom) [inline delly.wdl]: Memory needs to be string, like suggestions above.

File: delly.wdl [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+        String dockerImage = "quay.io/biocontainers/delly:0.8.1--h4037b6b_1"
+    }
+    
+
+    command <<<
+        set -e
+        mkdir -p $(dirname ~{outputPath})
```
Comment (JasperBoom) [inline delly.wdl]: Add quotations here like above.

File: delly.wdl [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+        delly call \
+        -o ~{outputPath} \
+        -g ~{referenceFasta} \
+        ~{bamFile}
+    >>>
+
+    output {
+        File dellyBcf = "~{outputPath}"
```
Comment (JasperBoom) [inline delly.wdl]: See comment above.

File: clever.wdl (hunk 5 above)
Comment (cagaser) [reply clever.wdl]: Ah I have to delete those

File: clever.wdl (hunk 3 above)
Comment (cagaser) [reply clever.wdl]: I thought we keep a default docker-image in each task?

File: clever.wdl [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+
+task Mateclever {
+    input {
+        File fiteredBam
+        File indexedFiteredBam
+        BwaIndex bwaIndex
+        File predictions
+        String outputPath
+        Int threads = 10 
+        String memory = 15
```
Comment (DavyCats) [inline clever.wdl]: `15` is not a string ;)

File: delly.wdl [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+import "common.wdl"
+
+task CallSV {
+    input {
+        File bamFile
+        File bamIndex
+        File referenceFasta
+        File referenceFastaFai
+        String outputPath        
+        String memory = 15
```
Comment (DavyCats) [inline delly.wdl]: Same comment as above

File: manta.wdl [hunk 18]
```diff
@@ -74,3 +74,47 @@ task Somatic {
...
+        File bamIndex
+        File referenceFasta
+        File referenceFastaFai
+        String runDir
+        File? callRegions
+        File? callRegionsIndex
+        Boolean exome = false
+        
+        Int cores = 1
+        String memory = 4
```
Comment (DavyCats) [inline manta.wdl]: Here as well

File: picard.wdl [hunk 19]
```diff
@@ -608,4 +610,35 @@ task SortVcf {
...
\ No newline at end of file
+}
+
+task RenameSample {
+    input {
+        String dockerImage = "quay.io/biocontainers/picard:2.19.0--0"
+        File inputVcf
+        String outputPath
+        String newSampleName
+        Int memory = 8
```
Comment (DavyCats) [inline picard.wdl]: This should be a string

File: picard.wdl [hunk 20]
```diff
@@ -608,4 +610,35 @@ task SortVcf {
...
+        Float memoryMultiplier = 3.0
```
Comment (DavyCats) [inline picard.wdl]: We don't use the memory multiplier anymore. Instead, provide two memory inputs:
- memory (which is given to runtime)
- javaXmx (which is given to java)

File: samtools.wdl [hunk 21]
```diff
@@ -307,6 +307,8 @@ task View {
         File inFile
         File? referenceFasta
         String outputFileName = "view.bam"
+	    Boolean? includeHeader
```
Comment (DavyCats) [inline samtools.wdl]:
```suggestion
	  Boolean? includeHeader
```

File: samtools.wdl [hunk 22]
```diff
@@ -307,6 +307,8 @@ task View {
...
+        Boolean? outputBam
         Boolean? uncompressedBamOutput
```
Comment (DavyCats) [inline samtools.wdl]: Boolean shouldn't really be optional, just give them a default value of `false`:
```suggestion
        Boolean uncompressedBamOutput = false
```

File: samtools.wdl [hunk 23]
```diff
@@ -363,3 +366,32 @@ task View {
...
+        String dockerImage = "quay.io/biocontainers/samtools:1.8--h46bd0b3_5"
+    }
+
+    command {
+        set -e
+        mkdir -p "$(dirname ~{outputPathBam})"
+        samtools view -h ~{bamFile} | \
+        awk 'length($10) > 30 || $1 ~/^@/' | \
+        samtools view -bS -> ~{outputPathBam} 
+        samtools index ~{outputPathBam}
```
Comment (DavyCats) [inline samtools.wdl]: The convention within miniwdl tends to be that BAM indexes only get the `.bai`, not `.bam.bai`. You should give an output path here as well.

File: survivor.wdl [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+        Array[File] filePaths
+        Int breakpointDistance = 1000
+        Int suppVecs = 2
+        Int svType = 1
+        Int strandType = 1
+        Int distanceBySvSize = 0
+        Int minSize = 30
+        String sample
+        String outputPath
+        String memory = 128
```
Comment (DavyCats) [inline survivor.wdl]: This is not a string

File: clever.wdl (hunk 3 above)
Comment (DavyCats) [reply clever.wdl]: We do. I think @JasperBoom just meant that the extra blank line should be removed

Comment (cagaser) [reply clever.wdl]: Ah ok! got it

File: picard.wdl (hunk 19 above)
Comment (cagaser) [reply picard.wdl]: But there is memory multiplier 
ceil(memory * memoryMultiplier)

File: clever.wdl [hunk 25]
```diff
@@ -0,0 +1,79 @@
...
+        ~{outputPath}
+    } 
+
+    output {
+        File predictions = outputPath + "/predictions.vcf"
+    }   
+    
+    runtime {
+        cpu: threads
+        memory: mem
```
Comment (JasperBoom) [inline clever.wdl]: "mem" here does not match "memory" in the inputs

**Comment by JasperBoom** [comment]
Don't forget your parameter_meta :wink:

File: clever.wdl [hunk 26]
```diff
@@ -0,0 +1,102 @@
...
+        bamFile: {description: "The bam file to process.", category: "required"}
+        bamIndex: {description: "The index bam file.", category: "required"}
+        outputPath: {description: "The location the output VCF file should be written.", category: "common"}
+        dockerImage: {description: "The docker image used for this task. Changing this may result in errors which the developers may choose not to address.",
+                      category: "advanced"}
+        bwaIndex: {description: "The BWA index files.", category: "required"}
+        memory: {description: "The memory required to run the programs", category: "common"}
+        threads: {description: "The the number of threads required to run a program", category: "common"}
+    }
+
```
Comment (JasperBoom) [inline clever.wdl]:
```suggestion
```

File: clever.wdl [hunk 27]
```diff
@@ -0,0 +1,102 @@
...
+        BwaIndex bwaIndex
+        String outputPath        
+        Int threads = 10 
+        String memory = "15G"
+        String dockerImage = "quay.io/biocontainers/clever-toolkit:2.4--py36hcfe0e84_6"
+    }   
+    
+    command { 
+        set -e
+        mkdir -p $(dirname ~{outputPath})
```
Comment (JasperBoom) [inline clever.wdl]:
```suggestion
        mkdir -p "$(dirname ~{outputPath})"
```

File: clever.wdl [hunk 28]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    parameter_meta {
+        # inputs
+        bamFile: {description: "The bam file to process.", category: "required"}
+        bamIndex: {description: "The index bam file.", category: "required"}
+        bwaIndex: {description: "The BWA index files.", category: "required"}
+        outputPath: {description: "The location the output VCF file should be written.", category: "common"}
+        threads: {description: "The the number of threads required to run a program", category: "common"}
+        memory: {description: "The memory required to run the programs", category: "common"}
```
Comment (DavyCats) [inline clever.wdl]: memory and threads inputs should be in the advanced category

File: clever.wdl [hunk 29]
```diff
@@ -0,0 +1,108 @@
...
+    parameter_meta {
+        # inputs
+        fiteredBam: {description: "The bam file where sequences less than 30bp were removed.", category: "required"}
+        indexedFiteredBam: {description: "The index of the filtered bam file.", category: "required"}
+        bwaIndex: {description: "The BWA index files.", category: "required"}
+        predictions: {description: "The predicted deletions (VCF) from clever.", category: "required"}
+        outputPath: {description: "The location the output VCF file should be written.", category: "common"}
+        cleverMaxDelLength: {description: "Maximum deletion length to look for from clever predictions.", category: "common"}
+        maxLengthDiff: {description: "Maximum length difference between split-read and read-pair deletion to be considered identical ", category: "common"}
+        maxOffset: {description: "Maximum center distance between split-read and read-pair deletion to be considered identical", category: "common"}
```
Comment (DavyCats) [inline clever.wdl]: These sound like advanced inputs, not common ones

File: bcftools.wdl [hunk 30]
```diff
@@ -0,0 +1,24 @@
...
+        bcftools view ~{bcf} -O v -o ~{outputPath}
+    }
+
+    output {
+        File outputVcf = outputPath
+    }
+
+    runtime {
+        docker: dockerImage
+    }
```
Comment (DavyCats) [inline bcftools.wdl]: parameter_meta is missing

File: manta.wdl [hunk 31]
```diff
@@ -2,6 +2,64 @@ version 1.0
...
+        bamIndex: {description: "The index bam file.", category: "required"}
+        referenceFasta: {description: "The reference fasta file also used for mapping.", category: "advanced"}
+        referenceFastaFai: {description: "Fasta index (.fai) file of the reference", category: "required" }
+        runDir: {description: "The directory to use as run/output directory.", category: "common"}
+        callRegions: {description: "The bed file which indicates the regions to operate on.", category: "common"}
+        callRegionsIndex: {description: "The index of the bed file which indicates the regions to operate on.", category: "common"}
+        exome: {description: "Whether or not the data is from exome sequencing.", category: "common"}
+        cores: {description: "The the number of cores required to run a program", category: "common"}
+        memoryGb: {description: "The memory required to run the manta", category: "common"}
+        dockerImage: {description: "The docker image used for this task. Changing this may result in errors which the developers may choose not to address.", category: "advanced"}
```
Comment (DavyCats) [inline manta.wdl]: These should be advanced not common

File: delly.wdl [hunk 32]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    parameter_meta {
+        # inputs
+        bamFile: {description: "The bam file to process.", category: "required"}
+        bamIndex: {description: "The index bam file.", category: "required"}
+        referenceFasta: {description: "The reference fasta file also used for mapping.", category: "advanced"}
+        referenceFastaFai: {description: "Fasta index (.fai) file of the reference", category: "required" }
+        outputPath: {description: "The location the output VCF file should be written.", category: "common"}
+        memory: {description: "The memory required to run the programs", category: "common"}
```
Comment (DavyCats) [inline delly.wdl]: should be advanced

File: picard.wdl [hunk 33]
```diff
@@ -1,5 +1,7 @@
 version 1.0
 
+import "common.wdl"
```
Comment (DavyCats) [inline picard.wdl]: What's this import for?

File: picard.wdl [hunk 34]
```diff
@@ -608,4 +619,45 @@ task SortVcf {
...
+        memory: memory
+    }
+
+    parameter_meta {
+        # inputs
+        inputVcf: {description: "The VCF file to process.", category: "required"}
+        outputPath: {description: "The location the output VCF file should be written.", category: "common"}
+        newSampleName: {description: "A string to replace the old sample name.", category: "required"}
+        memory: {description: "The memory required to run the programs", category: "common"}
+        javaXmx: {description: "The max. memory allocated for JAVA", category: "common"}
```
Comment (DavyCats) [inline picard.wdl]: Should be advanced

File: samtools.wdl [hunk 35]
```diff
@@ -307,7 +315,9 @@ task View {
         File inFile
         File? referenceFasta
         String outputFileName = "view.bam"
-        Boolean? uncompressedBamOutput
+	  Boolean? includeHeader
+        Boolean? outputBam
```
Comment (DavyCats) [inline samtools.wdl]: Booleans shouldn't be optional, use a default of false instead

File: samtools.wdl [hunk 36]
```diff
@@ -363,3 +375,31 @@ task View {
...
+    }
+
+    output {
+        File filteredBam = outputPathBam
+        File filteredBamIndex = outputPathBamIndex
+    }
+
+    runtime {
+        docker: dockerImage
+    }
```
Comment (DavyCats) [inline samtools.wdl]: missing parameter_meta

File: survivor.wdl [hunk 37]
```diff
@@ -0,0 +1,59 @@
+version 1.0
+
+import "common.wdl"
```
Comment (DavyCats) [inline survivor.wdl]: What's this import for?

File: survivor.wdl [hunk 38]
```diff
@@ -0,0 +1,59 @@
...
+        Int breakpointDistance = 1000
+        Int suppVecs = 2
+        Int svType = 1
+        Int strandType = 1
+        Int distanceBySvSize = 0
+        Int minSize = 30
+        String sample
+        String outputPath
+
+        String memory = "128G"
```
Comment (DavyCats) [inline survivor.wdl]: Does it really need this much memory?

File: survivor.wdl [hunk 39]
```diff
@@ -0,0 +1,59 @@
...
+        breakpointDistance: {description: "The distance between pairwise breakpoints between SVs", category: "required"}
+        suppVecs: {description: "The minimum number of SV callers to support the merging", category: "required"}
+        svType: {description: "A boolean to include the type SV to be merged", category: "required"}
+        strandType: {description: "A boolean to include strand type of an SV to be merged", category: "required"}
+        distanceBySvSize: {description: "A boolean to predict the pairwise distance between the SVs based on their size", category: "required"}
+        minSize: {description: "The mimimum size of SV to be merged", category: "required"}
+        sample: {description: "The name of the sample", category: "required"}
+        outputPath: {description: "The location the output VCF file should be written.", category: "common"}
+        memory: {description: "The memory required to run the programs", category: "common"}
+        dockerImage: {description: "The docker image used for this task. Changing this may result in errors which the developers may choose not to address.", category: "advanced"}
```
Comment (DavyCats) [inline survivor.wdl]: Please check the categories, most the ones noted as `required` should be `advanced`

File: samtools.wdl (hunk 35 above)
Comment (cagaser) [reply samtools.wdl]: I don't think I wrote this. I only added the task FilterShortReadsBam. 
Should I adjust it anyways?

File: survivor.wdl (hunk 38 above)
Comment (cagaser) [reply survivor.wdl]: I can't remember why I put it like that. I think It was when I was running WGS familial melanoma data.

File: samtools.wdl (hunk 35 above)
Comment (DavyCats) [reply samtools.wdl]: yes

File: bcftools.wdl [hunk 40]
```diff
@@ -0,0 +1,51 @@
...
+
+    parameter_meta {
+        bcf: {description: "The generated BCF from an SV caller", category: "advanced"}
+        outputPath: {description: "The location the output VCF file should be written.", category: "common"}
```
Comment (DavyCats) [inline bcftools.wdl]: These are required inputs.

File: clever.wdl [hunk 41]
```diff
@@ -0,0 +1,128 @@
...
+        memory: memory
+        docker: dockerImage
+    }
+
+    parameter_meta {
+        # inputs
+        bamFile: {description: "The bam file to process.", category: "required"}
+        bamIndex: {description: "The index bam file.", category: "required"}
+        bwaIndex: {description: "The BWA index files.", category: "required"}
+        predictions: {description: "The predicted deletions (VCF) from clever.", category: "advanced"}
```
Comment (DavyCats) [inline clever.wdl]: This is not an input. Please move it below the inputs and add a line like `# outputs` above it.

File: delly.wdl [hunk 42]
```diff
@@ -0,0 +1,65 @@
...
+        docker: dockerImage
+    }
+
+    parameter_meta {
+        # inputs
+        bamFile: {description: "The bam file to process.", category: "required"}
+        bamIndex: {description: "The index bam file.", category: "required"}
+        referenceFasta: {description: "The reference fasta file also used for mapping.", category: "required"}
+        referenceFastaFai: {description: "Fasta index (.fai) file of the reference", category: "required" }
+        outputPath: {description: "The location the output VCF file should be written.", category: "common"}
```
Comment (DavyCats) [inline delly.wdl]: This is a required input.

File: clever.wdl [hunk 43]
```diff
@@ -0,0 +1,128 @@
...
+        docker: dockerImage
+    }
+
+    parameter_meta {
+        # inputs
+        fiteredBam: {description: "The bam file where sequences less than 30bp were removed.", category: "required"}
+        indexedFiteredBam: {description: "The index of the filtered bam file.", category: "required"}
+        bwaIndex: {description: "The BWA index files.", category: "required"}
+        predictions: {description: "The predicted deletions (VCF) from clever.", category: "required"}
+        outputPath: {description: "The location the output VCF file should be written.", category: "common"}
```
Comment (DavyCats) [inline clever.wdl]: This is a required input.

File: manta.wdl [hunk 44]
```diff
@@ -20,7 +20,63 @@ version 1.0
...
+        bamFile: {description: "The bam file to process.", category: "required"}
+        bamIndex: {description: "The index bam file.", category: "required"}
+        referenceFasta: {description: "The reference fasta file also used for mapping.", category: "advanced"}
+        referenceFastaFai: {description: "Fasta index (.fai) file of the reference", category: "required" }
+        runDir: {description: "The directory to use as run/output directory.", category: "common"}
+        callRegions: {description: "The bed file which indicates the regions to operate on.", category: "common"}
+        callRegionsIndex: {description: "The index of the bed file which indicates the regions to operate on.", category: "common"}
+        exome: {description: "Whether or not the data is from exome sequencing.", category: "common"}
+        cores: {description: "The the number of cores required to run a program", category: "common"}
+        memoryGb: {description: "The memory required to run the manta", category: "common"}
```
Comment (DavyCats) [inline manta.wdl]: These should be advanced

File: manta.wdl [hunk 45]
```diff
@@ -20,7 +20,63 @@ version 1.0
...
+        docker: dockerImage
+    }
+
+    parameter_meta {
+        # inputs
+        bamFile: {description: "The bam file to process.", category: "required"}
+        bamIndex: {description: "The index bam file.", category: "required"}
+        referenceFasta: {description: "The reference fasta file also used for mapping.", category: "advanced"}
+        referenceFastaFai: {description: "Fasta index (.fai) file of the reference", category: "required" }
+        runDir: {description: "The directory to use as run/output directory.", category: "common"}
```
Comment (DavyCats) [inline manta.wdl]: This is required

File: picard.wdl [hunk 46]
```diff
@@ -628,4 +637,45 @@ task SortVcf {
...
+
+    runtime {
+        docker: dockerImage
+        memory: memory
+    }
+
+    parameter_meta {
+        # inputs
+        inputVcf: {description: "The VCF file to process.", category: "required"}
+        outputPath: {description: "The location the output VCF file should be written.", category: "common"}
```
Comment (DavyCats) [inline picard.wdl]: This is required

File: samtools.wdl [hunk 47]
```diff
@@ -383,3 +395,37 @@ task View {
                       category: "advanced"}
     }
 }
+
+task FilterShortReadsBam {
+    input {
+        File bamFile
+        String outputPathBam
+        String outputPathBamIndex
```
Comment (DavyCats) [inline samtools.wdl]: This should be based on the `outputPathBamIndex`, it shouldn't be an input.

File: samtools.wdl [hunk 48]
```diff
@@ -383,3 +395,37 @@ task View {
...
+    }
+
+    runtime {
+        docker: dockerImage
+    }
+
+    parameter_meta {
+        bamFile: {description: "The bam file to process.", category: "required"}
+        outputPathBam: {description: "The filtered bam file.", category: "advanced"}
+        outputPathBamIndex: {description: "The index of filtered bam file.", category: "advanced"}
```
Comment (DavyCats) [inline samtools.wdl]: These are required

File: survivor.wdl [hunk 49]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    runtime {
+        memory: memory
+        docker: dockerImage
+    }
+
+    parameter_meta {
+        # inputs
+        filePaths: {description: "An array of VCF files (predictions) to be merged by SURVIVOR", category: "advanced"}
```
Comment (DavyCats) [inline survivor.wdl]: This is required

File: survivor.wdl [hunk 50]
```diff
@@ -0,0 +1,76 @@
...
+        breakpointDistance: {description: "The distance between pairwise breakpoints between SVs", category: "advanced"}
+        suppVecs: {description: "The minimum number of SV callers to support the merging", category: "advanced"}
+        svType: {description: "A boolean to include the type SV to be merged", category: "advanced"}
+        strandType: {description: "A boolean to include strand type of an SV to be merged", category: "advanced"}
+        distanceBySvSize: {description: "A boolean to predict the pairwise distance between the SVs based on their size", category: "advanced"}
+        minSize: {description: "The mimimum size of SV to be merged", category: "advanced"}
+        outputPath: {description: "The location the output VCF file should be written.", category: "common"}
```
Comment (DavyCats) [inline survivor.wdl]: This is required

File: samtools.wdl (hunk 47 above)
Comment (cagaser) [reply samtools.wdl]: You mean outputPathBam?

File: bcftools.wdl (hunk 40 above)
Comment (cagaser) [reply bcftools.wdl]: The BCF was generated by Delly, isn't that advanced?

File: clever.wdl [hunk 51]
```diff
@@ -0,0 +1,130 @@
...
+
+import "bwa.wdl"
+
+task Mateclever {
+    input {
+        File fiteredBam
+        File indexedFiteredBam
+        BwaIndex bwaIndex
+        File predictions
+        String outputPath
```
Comment (rhpvorderman) [inline clever.wdl]: It is best to give `outputPath` a default. If a task runs in the cloud you should not have to worry about this.

File: bcftools.wdl [hunk 52]
```diff
@@ -0,0 +1,51 @@
...
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+task Bcf2Vcf {
+    input {
+        File bcf
+        String outputPath
```
Comment (rhpvorderman) [inline bcftools.wdl]: It is best to give outputPath a default. If a task runs in the cloud you should not have to worry about this.

File: clever.wdl [hunk 53]
```diff
@@ -0,0 +1,130 @@
...
+        dockerImage: {description: "The docker image used for this task. Changing this may result in errors which the developers may choose not to address.", category: "advanced"}
+    }
+}
+
+task Prediction {
+    input {
+        File bamFile
+        File bamIndex
+        BwaIndex bwaIndex
+        String outputPath
```
Comment (rhpvorderman) [inline clever.wdl]: It is best to give outputPath a default. If a task runs in the cloud you should not have to worry about this.

File: delly.wdl [hunk 54]
```diff
@@ -0,0 +1,65 @@
...
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+task CallSV {
+    input {
+        File bamFile
+        File bamIndex
+        File referenceFasta
+        File referenceFastaFai
+        String outputPath
```
Comment (rhpvorderman) [inline delly.wdl]: It is best to give outputPath a default. If a task runs in the cloud you should not have to worry about this.

File: manta.wdl [hunk 55]
```diff
@@ -20,7 +20,63 @@ version 1.0
...
 # SOFTWARE.
 
-import "common.wdl"
+task Germline {
+    input {
+        File bamFile
+        File bamIndex
+        File referenceFasta
+        File referenceFastaFai
+        String runDir
```
Comment (rhpvorderman) [inline manta.wdl]: It is best to give runDir a default. If a task runs in the cloud you should not have to worry about this.

File: picard.wdl [hunk 56]
```diff
@@ -628,4 +637,45 @@ task SortVcf {
...
                       category: "advanced"}
     }
-}
\ No newline at end of file
+}
+
+task RenameSample {
+    input {
+        File inputVcf
+        String outputPath
```
Comment (rhpvorderman) [inline picard.wdl]: Defaiult

File: samtools.wdl (hunk 47 above)
Comment (DavyCats) [reply samtools.wdl]: oh yes, sorry, should be based on `outputPathBam`

File: survivor.wdl [hunk 57]
```diff
@@ -0,0 +1,76 @@
...
+task Merge {
+    input{
+        Array[File] filePaths
+        Int breakpointDistance = 1000
+        Int suppVecs = 2
+        Int svType = 1
+        Int strandType = 1
+        Int distanceBySvSize = 0
+        Int minSize = 30
+        String outputPath
```
Comment (rhpvorderman) [inline survivor.wdl]: default needed

File: clever.wdl (hunk 51 above)
Comment (cagaser) [reply clever.wdl]: Like this right? `String outputPath = .  `

**Pull request merged**